File: src/main/java/ism/java/entity/Produit.java
package ism.java.entity;

import ism.java.enums.TypeProduit;
import ism.java.enums.TypeComplement;

public class Produit {
    
    private int id;
    private String nom;
    private double prix;
    private String image;
    private TypeProduit typeProduit;
    private TypeComplement typeComplement;
    private boolean estArchive;
    
    public Produit() {
        this.estArchive = false;
    }
    
    public Produit(int id, String nom, double prix, String image, 
                   TypeProduit typeProduit, TypeComplement typeComplement, boolean estArchive) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.image = image;
        this.typeProduit = typeProduit;
        this.typeComplement = typeComplement;
        this.estArchive = estArchive;
    }
   
    public Produit(String nom, double prix, String image) {
        this();
        this.nom = nom;
        this.prix = prix;
        this.image = image;
        this.typeProduit = TypeProduit.BURGER;
        this.typeComplement = null;
    }
    
    
    public Produit(String nom, double prix, String image, TypeComplement typeComplement) {
        this();
        this.nom = nom;
        this.prix = prix;
        this.image = image;
        this.typeProduit = TypeProduit.COMPLEMENT;
        this.typeComplement = typeComplement;
    }
    
    // Getters et Setters
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public String getNom() {
        return nom;
    }
    
    public void setNom(String nom) {
        this.nom = nom;
    }
    
    public double getPrix() {
        return prix;
    }
    
    public void setPrix(double prix) {
        this.prix = prix;
    }
    
    public String getImage() {
        return image;
    }
    
    public void setImage(String image) {
        this.image = image;
    }
    
    public TypeProduit getTypeProduit() {
        return typeProduit;
    }
    
    public void setTypeProduit(TypeProduit typeProduit) {
        this.typeProduit = typeProduit;
    }
    
    public TypeComplement getTypeComplement() {
        return typeComplement;
    }
    
    public void setTypeComplement(TypeComplement typeComplement) {
        this.typeComplement = typeComplement;
    }
    
    public boolean isEstArchive() {
        return estArchive;
    }
    
    public void setEstArchive(boolean estArchive) {
        this.estArchive = estArchive;
    }
    
    public boolean isBurger() {
        return typeProduit == TypeProduit.BURGER;
    }
    
    public boolean isComplement() {
        return typeProduit == TypeProduit.COMPLEMENT;
    }
    
    public boolean isBoisson() {
        return isComplement() && typeComplement == TypeComplement.BOISSON;
    }
    
    public boolean isFrite() {
        return isComplement() && typeComplement == TypeComplement.FRITE;
    }
}

File: src/Entity/Produit.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
#[ORM\Table(name: "produit")]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: "id", type: "integer")]
    private ?int $id = null;

    #[ORM\Column(name: "nom", type: "string", length: 255)]
    #[Assert\NotBlank(message: "Le nom du produit est obligatoire")]
    private string $nom = '';

    #[ORM\Column(name: "prix", type: "decimal", precision: 10, scale: 2)]
    #[Assert\PositiveOrZero(message: "Le prix doit être positif")]
    private float $prix = 0.0;

    #[ORM\Column(name: "image", type: "string", length: 500, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(name: "type_produit", type: "string", length: 50)]
    #[Assert\NotBlank]
    #[Assert\Choice(
        choices: ['BURGER', 'COMPLEMENT'],
        message: "Le type de produit doit être BURGER ou COMPLEMENT"
    )]
    private string $typeProduit = '';

    #[ORM\Column(name: "type_complement", type: "string", length: 50, nullable: true)]
    #[Assert\Choice(
        choices: ['FRITE', 'BOISSON'],
        message: "Le type de complément doit être FRITE ou BOISSON"
    )]
    private ?string $typeComplement = null;

    #[ORM\Column(name: "est_archive", type: "boolean")]
    private bool $estArchive = false;

  
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;
        return $this;
    }

    public function getPrix(): float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;
        return $this;
    }

    public function getPrixFormate(): string
    {
        return number_format($this->prix, 0, ',', ' ') . ' FCFA';
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): static
    {
        $this->image = $image;
        return $this;
    }

    public function getTypeProduit(): string
    {
        return $this->typeProduit;
    }

    public function setTypeProduit(string $typeProduit): static
    {
        $this->typeProduit = $typeProduit;
        return $this;
    }

    public function getTypeComplement(): ?string
    {
        return $this->typeComplement;
    }

    public function setTypeComplement(?string $typeComplement): static
    {
        $this->typeComplement = $typeComplement;
        return $this;
    }

    public function isEstArchive(): bool
    {
        return $this->estArchive;
    }

    public function setEstArchive(bool $estArchive): static
    {
        $this->estArchive = $estArchive;
        return $this;
    }

    
    public function isBurger(): bool
    {
        return $this->typeProduit === 'BURGER';
    }

    
    public function isComplement(): bool
    {
        return $this->typeProduit === 'COMPLEMENT';
    }

   
    public function isBoisson(): bool
    {
        return $this->isComplement() && $this->typeComplement === 'BOISSON';
    }

    
    public function isFrite(): bool
    {
        return $this->isComplement() && $this->typeComplement === 'FRITE';
    }

    public function __toString(): string
    {
        $type = $this->isBurger() ? 'Burger' : $this->typeComplement ?? 'Produit';
        return sprintf('%s (%s - %s)', $this->nom, $type, $this->getPrixFormate());
    }
}